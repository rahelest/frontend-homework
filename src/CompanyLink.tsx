import { Company } from './types'
import styled from '@emotion/styled'

const CompanyLink = ({ name }: Company) => (
  <Name>{name}</Name>
)

export default CompanyLink

const Name = styled.div`
  font-weight: bold;
  font-size: 16px;
  letter-spacing: -0.2px;
  padding: 12px 16px;
`
