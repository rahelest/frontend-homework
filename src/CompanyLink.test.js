import { shallow } from 'enzyme'
import { CompanyLink } from './CompanyLink'

describe('<CompanyLink />', () => {
  const setSelectedCompanyIdMock = jest.fn()
  const toggleDropdownMenuVisibilityMock = jest.fn()

  const render = (props) =>
    shallow(
      <CompanyLink
        company={{ id: 1, name: 'Dummy company' }}
        setSelectedCompanyId={setSelectedCompanyIdMock}
        toggleDropdownMenuVisibility={toggleDropdownMenuVisibilityMock}
        {...props}
      />,
    )

  it('renders name when company is not selected', () => {
    expect(render({ isSelected: false })).toMatchSnapshot()
  })

  it('renders link with checkbox when company is selected', () => {
    expect(render({ isSelected: true })).toMatchSnapshot()
  })

  it('calls setSelectedCompanyId and toggleDropdownMenuVisibility when company link is clicked', () => {
    const component = render({ isSelected: false })
    component.find('[data-test-company-link]').simulate('click')

    expect(setSelectedCompanyIdMock).toHaveBeenCalled()
    expect(toggleDropdownMenuVisibilityMock).toHaveBeenCalled()
  })
})
