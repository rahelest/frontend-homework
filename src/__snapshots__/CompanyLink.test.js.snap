// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompanyLink /> renders link 1`] = `
<Styled(div)>
  Dummy company
</Styled(div)>
`;
